test/test_type_promotion.py

```diff
@@ -175,12 +189,17 @@ def _get_test_tensor(self, device, dtype, remove_zeros=False):
         shape = [5, 5, 5]
         if dtype == torch.bool:
             tensor = torch.randint(int(remove_zeros), 2, shape, device=device, dtype=dtype)
-        elif dtype.is_floating_point:
+        elif dtype.is_floating_point or dtype.is_complex:
             # "_th_normal_ not supported on CPUType for Half" so simpler create and convert
             tensor = torch.randn(shape, device=device)
             tensor = tensor.to(dtype)
             if remove_zeros:
-                tensor[torch.abs(tensor) < 0.05] = 5
+                tensor_abs = torch.abs(tensor)
```

Comment:
why do you need to do this?  Isn't the absolute value of a complex tensor exactly what you want to check?

Reply:
torch.abs returns a complex tensor for complex tensor. Here the abs output is compared by 0.05(real number). such an operation is not defined for a complex tensor, so until we have the abs function return float for complex, the workaround way is to convert the abs out to float ...